pkg/kubelet/pod_workers.go

```diff
@@ -174,6 +174,15 @@ type PodWorkers interface {
 	// Intended for use by the kubelet config loops, but not subsystems, which should
 	// use ShouldPod*().
 	CouldHaveRunningContainers(uid types.UID) bool
+
+	// ShouldPodBeFinished returns false before the pod
```

Comment:
nit: It might be worth to clearify here this in terms of the positive case first... i.e. `ShouldPodBeFinished` returns true when the syncTerminatedPod completes successfully.

I would also add a comment above `IsPodKnownTerminated` to make clear the distinction between `IsPodKnownTerminated` and `ShouldPodBeFinished` (i.e. `IsPodKnownTerminated` returns true when `syncTerminatingPod` completes successfully, vs `ShouldPodBeFinished` returns true when `syncTerminatingPod`  completes successfully.

Reply:
Agree, I'd like to see that fixed.

Reply:
Improved the comments